packages/utils/plugins-helpers/src/federation.ts

```diff
@@ -172,30 +174,32 @@ export class ApolloFederation {
 
       const { resolvableKeyDirectives } = objectTypeFederationDetails;
 
-      if (resolvableKeyDirectives.length) {
-        const outputs: string[] = [`{ __typename: '${parentType.name}' } &`];
+      if (resolvableKeyDirectives.length === 0) {
+        return federationTypeSignature;
+      }
```

Comment:
Since we need to return `federationTypeSignature` instead of `parentTypeSignature` when there's all `@key` are `resolvable=false`, I'm changing to return early here to make it easier to read as well